### PR TITLE
[IMP] hr_skills : manager are able to see his subordinates's skill's history

### PR DESCRIPTION
**Pull request by mepe-odoo**
With this commit, even if you don't have right in Employee application,
you are able to see your subordinates's skill's history. This improvment
will help manager to have a better idea of his subordinates's improvment.

task: 3792519-
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_skills/security/hr_skills_security.xml [hunk 1]
```diff
@@ -51,6 +51,21 @@
...
+        <field name="model_id" ref="model_hr_employee_skill_log"/>
+        <field name="domain_force">[(1, '=', 1)]</field>
+        <field name="groups" eval="[(4, ref('hr.group_hr_user'))]"/>
+    </record>
+
+    <record id="hr_employee_skill_log_manager_rule" model="ir.rule">
+        <field name="name">Employee Skill History: employee's manager : read</field>
+        <field name="model_id" ref="model_hr_employee_skill_log"/>
+        <!-- <field name="domain_force">[(0, '=', 1)]</field> -->
+        <field name="domain_force">['|', ('employee_id.id', 'in', user.employee_ids.subordinate_ids.ids), ('employee_id.id', 'in', user.employee_ids.ids)]</field>
```
Comment (jugj-odoo) [inline addons/hr_skills/security/hr_skills_security.xml:65]: hello, i think we can achieve the same result using the child_of operator like this
`        <field name="domain_force">[('employee_id.id', 'child_of', user.employee_ids.ids)]</field>
`
without moving the code from org chart, but has to be tested first

Comment (mepe-odoo) [reply addons/hr_skills/security/hr_skills_security.xml:65]: but you will just see your direct subordinate not your indirect subordinate no ?

Comment (sofiagvaladze) [reply addons/hr_skills/security/hr_skills_security.xml:65]: no, with child off, you will see all the subordinates :bulb:

Comment (mepe-odoo) [reply addons/hr_skills/security/hr_skills_security.xml:65]: ok i will change that :ok_hand:

File: addons/hr_skills/static/src/fields/skills_one2many/skills_one2many.js [hunk 2]
```diff
@@ -22,6 +23,12 @@ export class SkillsListRenderer extends CommonSkillsListRenderer {
         onWillStart(async () => {
             const res = await this.orm.searchCount('hr.skill', []);
             this.anySkills = res > 0;
+            [this.user] = await this.orm.read("res.users", [user.userId], ["employee_ids"]);
+            this.IsHrUser = await user.hasGroup("hr.group_hr_user");
+            this.userSubordinates = await this.orm.search(
+                "hr.employee",
+                [["id", "child_of", this.user.employee_ids]]
+            );
```
Comment (jugj-odoo) [inline addons/hr_skills/static/src/fields/skills_one2many/skills_one2many.js:31]: we can make this more efficient like this, didn't test but should work
```suggestion
            this.is_manager = await this.orm.searchCount(
                "hr.employee",
                [["id", "child_of", this.user.employee_ids],
                 ["id", "=", this.env.model.root.data.id]]
            );
```

Comment (mepe-odoo) [reply addons/hr_skills/static/src/fields/skills_one2many/skills_one2many.js:31]: not possible :/ The OnWillStart is not trigger at each page changement

File: addons/hr_skills/security/hr_skills_security.xml [hunk 3]
```diff
@@ -51,6 +51,20 @@
         <field name="groups" eval="[(4,ref('base.group_user'))]"/>
     </record>
 
+    <record id="hr_employee_skill_log_hr_user_rule" model="ir.rule">
+        <field name="name">Employee Skill History:  : read</field>
```
Comment (Bertrand2) [inline addons/hr_skills/security/hr_skills_security.xml:55]: typo I guess?
```suggestion
        <field name="name">Employee Skill History: read</field>
```

File: addons/hr_skills/static/src/fields/skills_one2many/skills_one2many.js [hunk 4]
```diff
@@ -62,7 +70,15 @@ export class SkillsListRenderer extends CommonSkillsListRenderer {
     }
 
     get showTimeline() {
-        return !this.props.list.context.no_timeline;
+        return (this.SkillsRight) && !this.props.list.context.no_timeline;
```
Comment (Bertrand2) [inline addons/hr_skills/static/src/fields/skills_one2many/skills_one2many.js:73]: Any reason for the parenthesis? :)

File: addons/hr_skills/static/src/fields/skills_one2many/skills_one2many.js [hunk 5]
```diff
@@ -62,7 +70,15 @@ export class SkillsListRenderer extends CommonSkillsListRenderer {
...
+    }
+
+    get SkillsRight() {
+        let isSubordinate = false;
+        if (this.env.model.root.data.employee_id){
+            isSubordinate = this.userSubordinates.includes(this.env.model.root.data.employee_id[0])
```
Comment (Bertrand2) [inline addons/hr_skills/static/src/fields/skills_one2many/skills_one2many.js:79]:
```suggestion
        if (this.env.model.root.data.employee_id) {
            isSubordinate = this.userSubordinates.includes(this.env.model.root.data.employee_id[0]);
```

File: addons/hr_skills/static/src/fields/skills_one2many/skills_one2many.js [hunk 6]
```diff
@@ -62,7 +70,15 @@ export class SkillsListRenderer extends CommonSkillsListRenderer {
...
+        }
+        return (this.IsHrUser || isSubordinate )
```
Comment (Bertrand2) [inline addons/hr_skills/static/src/fields/skills_one2many/skills_one2many.js:81]:
```suggestion
        return this.IsHrUser || isSubordinate;
```

**Review by Bertrand2 — COMMENTED** [review]
NAU: Nitpicking As Usual :copyright: 
Otherwise LGTM! :)

**Comment by Bertrand2** [comment]
robodoo r+

**Pull request closed without merging**